src/renderers/dom/server/ReactDOMServerRendering.js

```diff
@@ -0,0 +1,710 @@
+    if (child == null) {
+      return;
+    }
+    if (typeof child === 'string' || typeof child === 'number') {
+      content += child;
+    } else if (!didWarnInvalidOptionChildren) {
+      didWarnInvalidOptionChildren = true;
+      warning(
+        false,
+        'Only strings and numbers are supported as <option> children.',
+      );
+    }
+  });
+  return content;
+}
+
+function maskContext(type, context) {
+  var contextTypes = type.contextTypes;
+  if (!contextTypes) {
+    return emptyObject;
+  }
+  var maskedContext = {};
+  for (var contextName in contextTypes) {
+    maskedContext[contextName] = context[contextName];
+  }
+  return maskedContext;
+}
+
+function checkContextTypes(typeSpecs, values, location: string) {
+  if (__DEV__) {
+    checkPropTypes(
+      typeSpecs,
+      values,
+      location,
+      'Component',
+      () => '', // ReactDebugCurrentFrame.getStackAddendum,
+    );
+  }
+}
+
+function processContext(type, context) {
+  var maskedContext = maskContext(type, context);
+  if (__DEV__) {
+    if (type.contextTypes) {
+      checkContextTypes(type.contextTypes, maskedContext, 'context');
+    }
+  }
+  return maskedContext;
+}
+
+var STYLE = 'style';
+var RESERVED_PROPS = {
+  children: null,
+  dangerouslySetInnerHTML: null,
+  suppressContentEditableWarning: null,
+};
+
+function isCustomComponent(tagName, props) {
+  return tagName.indexOf('-') >= 0 || props.is != null;
+}
+
+function createOpenTagMarkup(
+  tagVerbatim,
+  tagLowercase,
+  props,
+  makeStaticMarkup,
+  isRootElement,
+  domID,
+  instForDebug,
+) {
+  var ret = '<' + tagVerbatim;
+
+  for (var propKey in props) {
+    if (!props.hasOwnProperty(propKey)) {
+      continue;
+    }
+    var propValue = props[propKey];
+    if (propValue == null) {
+      continue;
+    }
+    if (!registrationNameModules.hasOwnProperty(propKey)) {
+      if (propKey === STYLE) {
+        propValue = CSSPropertyOperations.createMarkupForStyles(
+          propValue,
+          instForDebug,
+        );
+      }
+      var markup = null;
+      if (isCustomComponent(tagLowercase, props)) {
+        if (!RESERVED_PROPS.hasOwnProperty(propKey)) {
+          markup = DOMPropertyOperations.createMarkupForCustomAttribute(
+            propKey,
+            propValue,
+          );
+        }
+      } else {
+        markup = DOMPropertyOperations.createMarkupForProperty(
+          propKey,
+          propValue,
+        );
+      }
+      if (markup) {
+        ret += ' ' + markup;
+      }
+    }
+  }
+
+  // For static pages, no need to put React ID and checksum. Saves lots of
+  // bytes.
+  if (makeStaticMarkup) {
+    return ret;
+  }
+
+  if (isRootElement) {
+    ret += ' ' + DOMPropertyOperations.createMarkupForRoot();
+  }
+  ret += ' ' + DOMPropertyOperations.createMarkupForID(domID);
+  return ret;
+}
+
+function resolve(child, context) {
+  if (Array.isArray(child)) {
+    throw new Error('well that was unexpected');
```

Comment:
Oh, duh.. @spicyj, what happens when child is passed in as an array (in a post-fiber world). Was this before we thought about fragments, since your original diff was from over a year ago? :P